flutter_local_notifications/test/macos_flutter_local_notifications_test.dart

```diff
@@ -681,5 +688,95 @@ void main() {
         isMethodCall('getNotificationAppLaunchDetails', arguments: null)
       ]);
     });
+
+    test('initialize with providesAppNotificationSettings', () async {
+      const DarwinInitializationSettings macOSInitializationSettings =
+          DarwinInitializationSettings(
+        requestProvidesAppNotificationSettings: true,
+      );
+      const InitializationSettings initializationSettings =
+          InitializationSettings(macOS: macOSInitializationSettings);
+      await flutterLocalNotificationsPlugin.initialize(initializationSettings);
+      expect(log, <Matcher>[
+        isMethodCall(
+          'initialize',
+          arguments: <String, Object>{
+            'requestAlertPermission': true,
+            'requestSoundPermission': true,
+            'requestBadgePermission': true,
+            'requestProvisionalPermission': false,
+            'requestCriticalPermission': false,
+            'requestProvidesAppNotificationSettings': true,
+            'defaultPresentAlert': true,
+            'defaultPresentSound': true,
+            'defaultPresentBadge': true,
+            'defaultPresentBanner': true,
+            'defaultPresentList': true,
+            'notificationCategories': <String>[],
+          },
+        ),
+      ]);
+    });
+
+    test('requestPermissions with providesAppNotificationSettings', () async {
+      await flutterLocalNotificationsPlugin
+          .resolvePlatformSpecificImplementation<
+              MacOSFlutterLocalNotificationsPlugin>()
+          ?.requestPermissions(
+            alert: true,
+            badge: true,
+            sound: true,
+            providesAppNotificationSettings: true,
+          );
+      expect(log, <Matcher>[
+        isMethodCall(
+          'requestPermissions',
+          arguments: <String, Object>{
+            'sound': true,
+            'alert': true,
+            'badge': true,
+            'provisional': false,
+            'critical': false,
+            'providesAppNotificationSettings': true,
+          },
+        ),
+      ]);
+    });
+
+    test(
+      'check notification settings includes providesAppNotificationSettings',
+      () async {
+        await flutterLocalNotificationsPlugin
+            .resolvePlatformSpecificImplementation<
+                MacOSFlutterLocalNotificationsPlugin>()
+            ?.checkPermissions();
+        expect(log, <Matcher>[
+          isMethodCall('checkPermissions', arguments: null),
+        ]);
+      },
+    );
+
+    test('macOS requestPermissions with all settings requested', () async {
+      await flutterLocalNotificationsPlugin
+          .resolvePlatformSpecificImplementation<
+              MacOSFlutterLocalNotificationsPlugin>()!
+          .requestPermissions(
+            sound: true,
+            badge: true,
+            alert: true,
+            provisional: true,
+            critical: true,
```

Comment:
I missed this before but since the test description is `macOS requestPermissions with all settings requested`, it looks this is missing requesting passing in `providesAppNotificationSettings` as `true`. iOS test does this but not the macOS one